clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -3565,8 +3565,30 @@ void ItaniumRTTIBuilder::BuildVTablePointer(const Type *Ty) {
   // Check if the alias exists. If it doesn't, then get or create the global.
   if (CGM.getItaniumVTableContext().isRelativeLayout())
     VTable = CGM.getModule().getNamedAlias(VTableName);
-  if (!VTable)
-    VTable = CGM.getModule().getOrInsertGlobal(VTableName, CGM.Int8PtrTy);
+
+  // To generate valid device code SPIR-V module with virtual functions these
+  // global pointers should have global address space, not private as by default
+  auto VTableTy =
+      CGM.getLangOpts().SYCLIsDevice &&
+              CGM.getLangOpts().SYCLAllowVirtualFunctions &&
+              (VTableName == ClassTypeInfo || VTableName == SIClassTypeInfo)
+          ? CGM.Int8Ty->getPointerTo(
+                static_cast<unsigned>(LangAS::opencl_global))
```

Comment:
Should it be some stuff like `sycl_global`?
```suggestion
                static_cast<unsigned>(LangAS::sycl_global))
```

Reply:
llvm-spirv translator can't accept this value - `sycl_global.` I can add something like:
`static_cast<unsigned>(asOpenCLLangAS(LangAS::sycl_global)))`
if it's okay

Reply:
Cast will likely produce wrong result. Let's leave `opencl_global` for now. Tagging @MrSidims for awareness: we might want to enhance the translator to support SYCL address spaces

Reply:
> llvm-spirv translator can't accept this value - `sycl_global.`

I would expect `sycl_global` and `opencl_global` to have the same representation in LLVM IR with spir* target triple - addrspace(1). Is it different?

Reply:
I've got:
`@_ZTVN10__cxxabiv120__si_class_type_infoE = external addrspace(11) global ptr addrspace(11)`
with
`clang++ -cc1 -fsycl-allow-virtual-functions -fsycl-is-device -internal-isystem Inputs -emit-llvm simple-sycl-virtual-function.cpp -triple spir64-unknown-unknown`
Am I doing something wrong?

Reply:
I think this suggestion https://github.com/intel/llvm/pull/7255#discussion_r1016455745 should fix this issue.